Client/Frontend/Settings/SettingsContentViewController.swift

```diff
@@ -0,0 +1,219 @@
+    private var interstitialErrorView: UILabel!
+
+    // The web view that displays content.
+    var webView: WKWebView!
+
+    private func startLoading(timeout: Double = DefaultTimeoutTimeInterval) {
+        if self.isLoaded {
+            return
+        }
+        if timeout > 0 {
+            self.timer = NSTimer.scheduledTimerWithTimeInterval(timeout, target: self, selector: "SELdidTimeOut", userInfo: nil, repeats: false)
+        } else {
+            self.timer = nil
+        }
+        self.webView.loadRequest(NSURLRequest(URL: url))
+        self.interstitialSpinnerView.startAnimating()
+    }
+
+    init(backgroundColor: UIColor = UIColor.whiteColor(), title: NSAttributedString? = nil) {
+        interstitialBackgroundColor = backgroundColor
+        settingsTitle = title
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Don't extend under navigation bar.
+        self.edgesForExtendedLayout = UIRectEdge.None
+
+        // This background agrees with the web page background.
+        // Keeping the background constant prevents a pop of mismatched color.
+        view.backgroundColor = interstitialBackgroundColor
+
+        self.webView = makeWebView()
+        view.addSubview(webView)
+
+        // Destructuring let causes problems.
+        let ret = makeInterstitialViews()
+        self.interstitialView = ret.0
+        self.interstitialSpinnerView = ret.1
+        self.interstitialErrorView = ret.2
+
+        view.addSubview(interstitialView)
+        if debug {
+            // This lets you see the underlying webview, but not interact with it.
+            interstitialView.alpha = 0.5
+        }
+        layoutInnerViews()
+        startLoading()
+    }
+
+    func makeWebView() -> WKWebView {
+        let config = WKWebViewConfiguration()
+        let webView = WKWebView(
+            frame: CGRect(x: 0, y: 0, width: 1, height: 1),
+            configuration: config
+        )
+        webView.navigationDelegate = self
+        return webView
+    }
+
+    private func makeInterstitialViews() -> (UIView, UIActivityIndicatorView, UILabel) {
+        let view = UIView()
+
+        // Keeping the background constant prevents a pop of mismatched color.
+        view.backgroundColor = interstitialBackgroundColor
+
+        let spinner = UIActivityIndicatorView(activityIndicatorStyle: UIActivityIndicatorViewStyle.Gray)
+        view.addSubview(spinner)
+
+        let error = UILabel()
+        if let settingsTitle = settingsTitle {
+            error.text = NSLocalizedString("Could not connect to \(settingsTitle.string).", comment: "Error message that is shown in settings when there was a problem loading \(settingsTitle.string).")
+            error.textColor = UIColor.redColor() // Firefox Orange!
```

Comment:
This was in the previous class, will consult robin/darrin about the confusion (keep it red or change it to firefox orange)